### PR TITLE
Update comet-ms version in openms-thirdparty

### PR DESCRIPTION
**Pull request by jonasscheid**
Follow-up of #46567

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by jonasscheid** [comment]
@jpfeuffer @timosachsenberg Have you ever seen this before? Do we hit a space limit now? That's new. Environment should not be bigger than the previous one 🤔 

`.15:17:08 BIOCONDA INFO (ERR) [Mar 22 15:17:08] SERR [Mar 22 15:17:08] ERRO Task processing failed: API error (500): {"message":"write /usr/local/share/sirius-ms-5.8.2-0/app/sirius_rest_service-5.8.2-boot.jar: no space left on device"}
`

**Comment by jpfeuffer** [comment]
Yes. I pray for the day they throw out mulled container testing.

**Comment by jonasscheid** [comment]
How do we move forward?

**Comment by jonasscheid** [comment]
@bioconda/core Is there any workaround here?

**Comment by timosachsenberg** [comment]
@bgruening any idea?

**Comment by jpfeuffer** [comment]
Since bioconda has a hard limit on package size due to its container tests, the only two solutions that I see:
- move to conda-forge
- reduce the size (e.g. by removing Sirius)

**Comment by timosachsenberg** [comment]
I would say we can remove Sirius.

**Comment by jpfeuffer** [comment]
It's quite suboptimal since the same minor version will have different thirdparty tools but we have no choice if bioconda can't even handle around 1GB of packages that don't even have to be tested since openms-thirdparty is just an agglomeration of packages basically.
And from my experience, I don't think anything will change soon.

**Comment by jpfeuffer** [comment]
The openms-thirdparty recipe has 0 bytes though. It just wraps versions of dependencies  that should be tested to work together.

**Comment by jonasscheid** [comment]
After removing Sirius it looks like there is some sort of hick-up when building 🤔

**Comment by jpfeuffer** [comment]
Try to push some comment somewhere to see if the comment goes away.

**Comment by jonasscheid** [comment]
Seems a persistent error. @jpfeuffer @timosachsenberg Do you think we can outsource this bundle to [multi-package-containers](https://github.com/BioContainers/multi-package-containers) ?

**Comment by jpfeuffer** [comment]
Sure, but
a) the current error seems to be during the build of the regular openms package (probably stalling because out of RAM or something)
b) multi-package container will not give you a conda package if you need that, too

**Comment by bgruening** [comment]
Work is still ongoing to remove mamba, support the new smaller package formats etc ... so I try restarting this build here.
What strikes me is that this does not to be a space issue anymore? Is this hitting now some other limitation (memory, runtime)?

**Comment by poshul** [comment]
Not sure what's going on with the CI since the logs have timed out @bgruening can you try to trigger the tests again?

**Comment by timosachsenberg** [comment]
##[error]No hosted parallelism has been purchased or granted. To request a free parallelism grant, please fill out the following form https://aka.ms/azpipelines-parallelism-request

**Comment by bernt-matthias** [comment]
> I guess this is a bioconda issue? when will this be resolved

From the bioconda gitter channel: _We are in contact with the CI provider, communication is slow. We will get back here if this is resolved._

**Comment by bgruening** [comment]
Circle-CI is back :)

File: recipes/openms-meta/meta.yaml [hunk 1]
```diff
@@ -143,11 +143,10 @@ outputs:
     requirements:
      run:
        - {{ pin_subpackage('openms', exact=True) }}
-       - comet-ms ==2023010
+       - comet-ms ==2023012
        - luciphor2 ==2020_04_03
        - msgf_plus ==2023.01.1202
        - percolator ==3.5
```
Comment (bgruening) [inline recipes/openms-meta/meta.yaml:149]:
```suggestion
       - percolator ==3.5
       - sirius-ms ==5.8.2
```

**Comment by jonasscheid** [comment]
Hooray! 🥳

**Comment by bgruening** [comment]
Then let's ship it :)

Thanks for your patience ...

**Pull request merged**